numpy/core/src/npysort/quicksort.c.src

```diff
@@ -107,6 +111,7 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
                 *sptr++ = pi - 1;
                 pl = pi + 1;
             }
+            *psdepth++ = cdepth + 1;
```

Comment:
every stack push does increase the depth, we are only skipping the increase in the shorter partition as it is not necessary to keep track of it there

the depth is never decreased the only thing that is done is switch to a different depth in a different "recursive" calltree